python/graphstorm/trainer/glem_np_trainer.py

```diff
@@ -158,17 +158,20 @@ def fit(self, train_loader, num_epochs,
             t0 = time.time()
             rt_profiler.start_record()
 
-            if freeze_input_layer_epochs <= epoch:
-                self._model.lm.unfreeze_input_encoder()
-                no_pl = False
-
             use_gnn = self._model.em_order_gnn_first
             # `use_gnn`` determines which part to train, if `em_order_gnn_first`
             # 1st round: train GNN, fix LM; 2nd round: train LM fix gnn
             for _ in range(2):
                 stage_start_time = time.time()
                 part_to_train = 'gnn' if use_gnn else 'lm'
-                self._model.toggle(part_to_train)
+                self._model.toggle(part_to_train, data)
+                if part_to_train == 'lm':
+                    # when training lm, unfreeze LM if past warmup epochs
+                    # otherwise, only train the LM decoder
+                    if freeze_input_layer_epochs <= epoch:
+                        no_pl = False
+                        self._model.lm.unfreeze_input_encoder()
```

Comment:
`toggle` doesn't unfreeze the LM model?

Reply:
Unfreezing of LM models is handled in trainer. Because the `freeze_input_layer_epochs` can decide whether to unfreeze LM. If still in warm up epochs, do not unfreeze LM.